homeronet/Plugin/Reddit.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using homeronet.Messages;
+using homeronet.Utility;
+using Newtonsoft.Json.Linq;
+
+namespace homeronet.Plugin
+{
+    public class Reddit : IPlugin
+    {
+        private class RedditCommand
+        {
+            public List<string> Subreddits { get; set; }
+            public Func<string, string> FormatMessage { get; set; }
+        }
+
+        private UriWebClient _webClient;
+        private Dictionary<string, RedditCommand> _commandForName = new Dictionary<string, RedditCommand>()
+        {
+            {
+                "alligator", new RedditCommand()
+                {
+                    Subreddits =  new List<string>() { "britishproblems" },
+                    FormatMessage =  message => "<alligator> " + message.ToLower()
+                }
+            },
+            {
+                "danl", new RedditCommand()
+                {
+                    Subreddits =  new List<string>() { "TheRedPill", "seduction", "CuckoldCommunity", "BikePorn", "electronic_cigarette", "snooker" },
+                    FormatMessage = message => "<bill`gate> " + message.ToLower()
+                }
+            },
+            {
+                "hurt", new RedditCommand()
+                {
+                    Subreddits = new List<string>() { "Buddhism", "explainlikeimfive", "gifs", "Eve", "weightlifting", "MensRights", "hardbodies", "DoesAnybodyElse" },
+                    FormatMessage = message => "<Hurt> " + message
+                }
+            }
+        };
+
+        public void Startup()
+        {
+            _webClient = new UriWebClient();
```

Comment:
Using UriWebClient when the request URI is not required. Stick to the base WebClient class. Non-failing.